lib/std/ascii.zig

```diff
@@ -47,214 +51,171 @@ pub const control_code = struct {
-    };
-    const graph = [_]u1{
-        //  0, 1, 2, 3, 4, 5, 6, 7 ,8, 9,10,11,12,13,14,15
-        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
-        0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
-        0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
-        1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
-
-        1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
-        1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
-        1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,
-        1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0,
+    const Index = enum {
+        control,
+        alphabetic,
+        hexadecimal,
+        alphanumeric,
+        whitespace,
     };
 
-    comptime var i = 0;
-    inline while (i < 128) : (i += 1) {
-        table[i] =
-            @as(u8, alpha[i]) << @enumToInt(tIndex.Alpha) |
-            @as(u8, hex[i]) << @enumToInt(tIndex.Hex) |
-            @as(u8, space[i]) << @enumToInt(tIndex.Space) |
-            @as(u8, digit[i]) << @enumToInt(tIndex.Digit) |
-            @as(u8, lower[i]) << @enumToInt(tIndex.Lower) |
-            @as(u8, upper[i]) << @enumToInt(tIndex.Upper) |
-            @as(u8, punct[i]) << @enumToInt(tIndex.Punct) |
-            @as(u8, graph[i]) << @enumToInt(tIndex.Graph);
+    /// Generates a table which is filled with the results of the given function for all characters.
+    fn getBoolTable(comptime condition: fn (u8) bool) [128]bool {
+        @setEvalBranchQuota(2000);
+        comptime var table: [128]bool = undefined;
+        comptime var index = 0;
+        while (index < 128) : (index += 1) {
+            table[index] = condition(index);
+        }
+        return table;
     }
-    mem.set(u8, table[128..256], 0);
-    break :init table;
-};
 
-fn inTable(c: u8, t: tIndex) bool {
-    return (combinedTable[c] & (@as(u8, 1) << @enumToInt(t))) != 0;
-}
+    fn init() CombinedTable {
+        comptime var table = [_]u8{0} ** 256;
+
+        const control_table = comptime getBoolTable(isControlNaive);
+        const alpha_table = comptime getBoolTable(isAlphabeticNaive);
+        const hex_table = comptime getBoolTable(isHexadecimalNaive);
+        const alphanumeric_table = comptime getBoolTable(isAlphanumericNaive);
+        const whitespace_table = comptime getBoolTable(isWhitespaceNaive);
+
+        comptime var i = 0;
+        inline while (i < 128) : (i += 1) {
+            table[i] =
+                @boolToInt(control_table[i]) << @enumToInt(Index.control) |
+                @boolToInt(alpha_table[i]) << @enumToInt(Index.alphabetic) |
+                @boolToInt(hex_table[i]) << @enumToInt(Index.hexadecimal) |
+                @boolToInt(alphanumeric_table[i]) << @enumToInt(Index.alphanumeric) |
+                @boolToInt(whitespace_table[i]) << @enumToInt(Index.whitespace);
+        }
 
-pub fn isAlNum(c: u8) bool {
-    return (combinedTable[c] & ((@as(u8, 1) << @enumToInt(tIndex.Alpha)) |
-        @as(u8, 1) << @enumToInt(tIndex.Digit))) != 0;
-}
```

Comment:
Original code also contains `mem.set(u8, table[128..256], 0);`.

Reply:
Yes, this is no longer required because of the way `table` is initialized: `[_]u8{0} ** 256`. Now that I looked at it closer though I realize that this same `mem.set` would still work so I will use the old way again.